iOS/Inventory/DittoManager.swift

```diff
@@ -79,13 +79,14 @@ extension DittoManager {
 
     func subscribeAllInventoryItems() {
 
-        let query = collections.inventories.findAll()
-
-        subscriptions.append(
-            query.subscribe()
-        )
+        do {
+            subscriptions.append(try ditto.sync.registerSubscription(query: "SELECT * FROM inventories"))
+        } catch {
+            print("Query Error: \(error)")
+        }
+        
         liveQueries.append(
-            query.observeLocal { [weak self] docs, event in
+            collections.inventories.findAll().observeLocal { [weak self] docs, event in
```

Comment:
Even though we may have to manually write deep diffing since v1 DQL doesn't return diffs, we would prefer to switch this line to DQL so that this could be the example to customers.

Reply:
Per the documentation, ["DQL does not support the Counter type for queries or modifications. Documents with the counter type will still continue to load but users will not be able to update or query for the counter field/value."](https://www.notion.so/getditto/Guide-for-Legacy-Builder-Query-DQL-Internal-de5eaa5a8d2246c7a3128ab42cb9966d#:~:text=DQL%20does%20not%20support%20the%20Counter,query%20for%20the%20counter%20field/value.)

https://www.notion.so/getditto/Guide-for-Legacy-Builder-Query-DQL-Internal-de5eaa5a8d2246c7a3128ab42cb9966d#:~:text=DQL%20does%20not%20support%20the%20Counter,query%20for%20the%20counter%20field/value.

Reply:
You're right - the callback result `DittoQueryResult` doesn't have methods to cast a value to a counter, so extracting a counter value (that is created by the builder API) from a collection may not be possible. It means we can't migrate observes to DQL unless we give up on using counters and use maps instead. FYI @skylerjokiel  @magicwave @okdistribute @radazzouz

Reply:
I am not sure what you mean by using maps - do you mean aggregates over documents? An embedded map would not not scalable for a counter.

Reply:
I agree that using maps to count numbers is not very reliable/scalable. I'd like to get opinions from @skylerjokiel about DQL design plans for it.